File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.py
class Solution:
    def partitionString(self, s: str) -> int:
        res=1
        i=0
        j=0
        r=''
        while j<len(s):
            if s[j] in r:
                r=''
                res+=1
                c=0
                j-=1
            else:
                r+=s[j]
            j+=1
        return res
        
                
            
                

File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.cpp
class Solution {
public:
    int partitionString(string s) {
        int i=0,res=1;
        set<char>h;
        for(i=0;i<s.length();i++){
            if(h.find(s[i])!=h.end()){
                res++;
                h.clear();
            }
            h.insert(s[i]);
        }
        return res;
    }
};